### ISSUE TITLE
Traverse entry points if some Dataset.<method> is not found

### ISSUE THREAD

**Issue by yarikoptic** [datalad/datalad#5990]
Observed by @jwodder while working on https://github.com/datalad/datalad-fuse/pull/3#issuecomment-921822862

Even though we declared interfaces to be added via entry points, and e.g. here if we move some functionality to -deprecated (BTW -- we need a release since `publish` move was not released in datalad-deprecated, known: https://github.com/datalad/datalad-deprecated/issues/29)

```shell
(git)lena:~datalad/datalad-master[master]git
$> python -c 'from datalad.distribution.dataset import Dataset; ds=Dataset("."); print(ds.publish)'     
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/yoh/proj/datalad/datalad-master/datalad/distribution/dataset.py", line 210, in __getattr__
    return super(Dataset, self).__getattribute__(attr)
AttributeError: 'Dataset' object has no attribute 'publish'
```

<details>
<summary>but it works if we import `datalad.api` as well, which then traverses the entry points:</summary> 

```shell
$> python -c 'from datalad.distribution.dataset import Dataset; from datalad.api import publish; ds=Dataset("."); print(ds.publish)'
<bound method Publish.__call__ of Dataset('/home/yoh/proj/datalad/datalad-deprecated')>
```
</details>

so I think we should add to  `Dataset.__getattr__` we already have traversal of entry points (code should be reused or RFed so not to be needed as triggered in datalad.api support code).

**Comment by yarikoptic** [comment]
@mih et al @datalad/developers WDYT?

**Comment by yarikoptic** [comment]
addressed by #5999 merged into maint
